### PR TITLE
chore: build drone YAML via jsonnet

### PR DESCRIPTION
**Pull request by andrewrynhard**
This PR aims to DRY the drone config file by using Jsonnet to generate
it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -196,50 +181,71 @@ talos-azure:
...
+.PHONY: iso
+iso:
+	@docker run --rm -i -v $(PWD)/build:/out autonomy/installer:$(TAG) iso
+
+.PHONY: container
+container: buildkitd
 	@$(BINDIR)/buildctl --addr $(BUILDKIT_HOST) \
 		build \
-		--output type=docker,dest=build/$@.tar,name=docker.io/autonomy/$@:$(TAG) \
+		--output type=docker,dest=build/$@.tar,name=docker.io/autonomy/talos:$(TAG) \
```
Comment (bradbeam) [inline Makefile]: just to verify, `dest=build/$@.tar` ( build/container.tar? ) is what we want, not `talos.tar`?

File: Makefile [hunk 2]
```diff
@@ -196,50 +181,71 @@ talos-azure:
...
-.PHONY: e2e
+.PHONY: e2e-integration
 e2e-integration:
 	@TAG=$(TAG) ./hack/test/$@.sh
 
-.PHONY: test
-test: buildkitd
+.PHONY: unit-tests
+unit-tests: buildkitd
+	@mkdir -p build
```
Comment (bradbeam) [inline Makefile]: doing `mkdir build` for unit tests seems unnecessary?

File: hack/drone.jsonnet [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+local conformance = e2e_setup + [
+  conformance_azure,
+  conformance_gce,
+];
+
+
+local release_step ={
+  name: "release",
+  image: "plugins/github-release",
+  settings:{
```
Comment (bradbeam) [inline hack/drone.jsonnet]: did you want to try to include the `note: ` bit to automatically pull in the ami names?

File: hack/drone.jsonnet [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+
+// CAPI.
+
+local push_image_azure = step("push-image-azure", [image_azure], creds_env_vars);
+local push_image_gce = step("push-image-gce", [image_gce], creds_env_vars);
+
+// Tests
+
+local lint = step("lint", [rootfs]);
```
Comment (bradbeam) [inline hack/drone.jsonnet]: why move lint to depend on rootfs and not clone?

File: hack/drone.jsonnet [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+// Steps
+
+local apps = [
+    machined,
+    osd,
+    trustd,
+    proxyd,
+    ntpd,
+    osctl_linux,
+    osctl_darwin,
```
Comment (bradbeam) [inline hack/drone.jsonnet]: nit, but having osctl_darwin here causes it to get built during e2e tests which isn't needed. not super significant and probably causes more complexity to filter it out

File: Makefile (hunk 1 above)
Comment (andrewrynhard) [reply Makefile]: yeah, every time I have to explain to people that `talos.tar` is the container. this is to just make it more clear as an artifact that its the container.

File: Makefile (hunk 2 above)
Comment (andrewrynhard) [reply Makefile]: Not sure why that is there TBH. Its nothing new, just a rename of `test` to `unit-test`. We can probably remove it.

File: hack/drone.jsonnet (hunk 3 above)
Comment (andrewrynhard) [reply hack/drone.jsonnet]: Shimmying that in to the way the `step` function works made things super ugly. In the `step` function I'm hoping to force us to use a single command for _most_ things. I found that it makes the steps a lot more clear and helps align the `Makefile` and `drone.jsonnet` file.

File: hack/drone.jsonnet (hunk 4 above)
Comment (andrewrynhard) [reply hack/drone.jsonnet]: I was thinking that we could group all the "tests" to run at the same time, but its probably better to just depend on `clone`. I will update.

File: hack/drone.jsonnet (hunk 5 above)
Comment (andrewrynhard) [reply hack/drone.jsonnet]: Yeah, it really makes the implementation uglier if we try to pull that bit out. I tried to design in in such a way that each pipeline sort of builds on the previous one. Plus this gets us closer to running a more complete e2e test.

File: hack/drone.jsonnet (hunk 3 above)
Comment (andrewrynhard) [reply hack/drone.jsonnet]: Plus with everything we talked about today in regard to publish and AWS asset instead of the AMIs, this felt less important.

**Comment by andrewrynhard** [comment]
I wanna iterate on this over the weekend and see how it performs in real life. Gonna merge, and follow up with some more cleanup as I figure out the best way to structure the Jsonnet code.

**Pull request merged**